hibernate-core/src/main/java/org/hibernate/loader/ast/internal/LoaderSelectBuilder.java

```diff
@@ -730,7 +732,14 @@ private void applyOrdering(TableGroup tableGroup, OrderByFragment orderByFragmen
 				fetchablePath = fetchParent.resolveNavigablePath( fetchable );
 			}
 
-			FetchTiming fetchTiming = fetchable.getMappedFetchOptions().getTiming();
+			FetchTiming fetchTiming;
+			if ( fetchParent instanceof EmbeddableFetchImpl && fetchable instanceof BasicAttributeMapping ) {
```

Comment:
Shouldn't we change the fetch timing in the mapping model already if this really is something that we can't support? I guess there are other parts that rely on the default timing as well that would otherwise need similar checks. Apart from that, what's the reason for this? I was of the impression that we do support lazy basic attributes in embeddables as Steve was specially handling this in the initializer for embeddables AFAIR.

Reply:
I moved the fetch timing determination in the mapping model. I do not think we support lazy loading of basic attributes for embeddables but I may be wrong